homeassistant/components/text/__init__.py

```diff
@@ -190,7 +194,20 @@ def max(self) -> int:
         return min(self.native_max, MAX_LENGTH_STATE_STATE)
 
     @property
-    def pattern(self) -> re.Pattern | None:
+    @final
+    def pattern_cmp(self) -> re.Pattern | None:
+        """Return a compiled pattern."""
+        if self.pattern is None and self.__pattern_cmp is not None:
+            self.__pattern_cmp = None
```

Comment:
For code style, you could return early here, and unindent below.
I doubt the cost of setting the internal attribute to `None` is much bigger than the check for `None`:
```suggestion
        if self.pattern is None:
            self.__pattern_cmp = None
            return None
```